### DISCUSSION TITLE
Advice on how to get feedback that a job hasn't run due to DI

### DISCUSSION THREAD

**Discussion by nrkirby** [quartznet/quartznet#1605] [General]
In our .NET app we've noticed that jobs with dependency injection issues will cause the job to **silently fail** (i.e. not run).

For example:

A job is registered as follows:

```
services.AddQuartz(q =>
        {
            q.UseMicrosoftDependencyInjectionJobFactory();
            q.ScheduleJob<DoSomethingJob>(trigger =>
                trigger.WithSimpleSchedule(builder =>
                    builder.WithIntervalInSeconds(5)
                        .RepeatForever()
                )
            );
       }

```

The job has a dependency that isn't wired up correctly with Microsoft dependency injection means that the job will **silently** not run.

Is there any guidance on how we can get feedback that there's a problem starting the job? Failing is fine but we need to know about it.

**Comment by lahma** [comment]
I think instantiation problems are logged by quartz with error level. Can you check the logs?

**Comment by nrkirby** [comment]
Is there a way for us to detect this without checking logs? i.e. exceptions thrown or similar?

We need a way to get an alert or get notified in some way that the job hasn't run.

**Reply by lahma** [reply]
Quartz has retries and compensation based on Execute method. Maybe it would be easier to create some integration tests to validate configuration is valid. I think MS DI has also switch to validate configuration.
